include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_consistent_splitting.cpp

```diff
@@ -0,0 +1,895 @@
+
+  AssertThrow(velocity[0].l2_norm() < 1.e-15 and pressure[0].l2_norm() < 1.e-15,
+              dealii::ExcMessage("Solution vector has to be zero for this stability analysis."));
+
+  AssertThrow(dealii::Utilities::MPI::n_mpi_processes(this->mpi_comm) == 1,
+              dealii::ExcMessage("Number of MPI processes has to be 1."));
+
+  std::cout << std::endl << "Analysis of eigenvalue spectrum:" << std::endl;
+
+  unsigned int const size = velocity[0].locally_owned_size();
+
+  dealii::LAPACKFullMatrix<Number> propagation_matrix(size, size);
+
+  // loop over all columns of propagation matrix
+  for(unsigned int j = 0; j < size; ++j)
+  {
+    // set j-th element to 1
+    velocity[0].local_element(j) = 1.0;
+
+    // solve time step
+    this->do_timestep_solve();
+
+    // dst-vector velocity_np is j-th column of propagation matrix
+    for(unsigned int i = 0; i < size; ++i)
+    {
+      propagation_matrix(i, j) = velocity_np.local_element(i);
+    }
+
+    // reset j-th element to 0
+    velocity[0].local_element(j) = 0.0;
+  }
+
+  // compute eigenvalues
+  propagation_matrix.compute_eigenvalues();
+
+  double norm_max = 0.0;
+
+  std::cout << "List of all eigenvalues:" << std::endl;
+
+  for(unsigned int i = 0; i < size; ++i)
+  {
+    double norm = std::abs(propagation_matrix.eigenvalue(i));
+    if(norm > norm_max)
+      norm_max = norm;
+
+    // print eigenvalues
+    std::cout << std::scientific << std::setprecision(5) << propagation_matrix.eigenvalue(i)
+              << std::endl;
+  }
+
+  std::cout << std::endl << std::endl << "Maximum eigenvalue = " << norm_max << std::endl;
+}
+
+template<int dim, typename Number>
+void
+TimeIntBDFConsistentSplitting<dim, Number>::do_timestep_solve()
+{
+  // pre-computations
+  pde_operator->interpolate_velocity_dirichlet_bc(velocity_dbc_np, this->get_next_time());
+
+  pressure_step();
+
+  momentum_step();
+
+  if(this->param.apply_penalty_terms_in_postprocessing_step)
+    penalty_step();
+
+  // evaluate convective term once the final solution at time
+  // t_{n+1} is known in the explicit case
+  if(this->param.convective_problem() and
+     this->param.treatment_of_convective_term == TreatmentOfConvectiveTerm::Explicit)
+    evaluate_convective_term();
+}
+
+template<int dim, typename Number>
+void
+TimeIntBDFConsistentSplitting<dim, Number>::evaluate_convective_term()
+{
+  dealii::Timer timer;
+  timer.restart();
+
+  pde_operator->evaluate_convective_term(this->convective_term_np,
+                                         velocity_np,
+                                         this->get_next_time());
+
+  this->timer_tree->insert({"Timeloop", "Convective step"}, timer.wall_time());
+}
+
+template<int dim, typename Number>
+void
+TimeIntBDFConsistentSplitting<dim, Number>::pressure_step()
+{
+  dealii::Timer timer;
+  timer.restart();
+
+  // compute right-hand-side vector
+  VectorType rhs(pressure_np);
+  rhs_pressure(rhs);
```

Comment:
As above, do not not create `rhs` from the vector `pressure_np` unless you actually want to use the content. Otherwise, do
```suggestion
  VectorType rhs;
  rhs.reinit(pressure_np);
  rhs_pressure(rhs);
```
if you want to have `rhs` initialized to zero, or if the zeroing happens inside `rhs_pressure` do `rhs.reinit(pressure_np, true);` where `true` indicates that we omit initializing the entries.